### ISSUE TITLE
Weird Finger Vertices

### ISSUE THREAD

**Issue by DFliyerz** [Frooxius/NeosPublic#856]
I mentioned this issue in the Discord, so I'll summarize it quickly:

While importing an avatar I had an issue where a vertex was popping off one of the fingers, and two of the fingers were way too close together. This issue doesn't show up in VRChat with the same model, so it might be an issue with how Neos is importing the mesh. Attached are screenshots of the hand in Neos, VRChat, and the fbx for the model.
![20200805080326_1](https://user-images.githubusercontent.com/8572440/89476908-be5f0b00-d740-11ea-810d-3fab7e259e3a.jpg)
![VRChat_1920x1080_2020-08-05_08-19-07 929](https://user-images.githubusercontent.com/8572440/89476919-c74fdc80-d740-11ea-8397-49ee231f5e2a.png)
[default.zip](https://github.com/Frooxius/NeosPublic/files/5031846/default.zip)

**Comment by TehTurk** [comment]
Going off the photos, This is due to the mesh having either loose Vertices or a vertices isn't weighted enough, VRC for comparison won't really help as typically they let you get away with this. There's like a hard limit for vertices like this but I can't recall how low the weights are allowed to go because the closer you get to zero the more floating point errors you get. I recommend just fixing them by adding some weight to them if you can. As for the fingers being in different spots, Honestly that could be just the difference in the skeletal models as well too.  Sorry I didn't get to address this in the discord.

**Comment by H3BO3** [comment]
The same issues don't occur in VRChat, so if the same model can be sent in, perhaps there's something that can be tweaked in the importer.

**Comment by TehTurk** [comment]
https://discord.com/channels/402159838827905024/439422161657135106/716525685396602911

**Comment by Frooxius** [comment]
I've looked into this, but this is a problem with the model itself, it has incorrect weight painting. It happens when it's imported into Unity as well, which uses the official FBX SDK by Autodesk:
![image](https://user-images.githubusercontent.com/8838625/89584868-1713ce00-d83d-11ea-992e-af3710d3f73f.png)

The incorrect weight paint is on the second bone weight. If the quality of the skinned mesh renderer is reduced all the way to 1 bone, it goes away:
![image](https://user-images.githubusercontent.com/8838625/89584911-2c88f800-d83d-11ea-91a9-058d45bc75b7.png)

This might be why you see a difference between VRChat and and Neos, as by default Neos uses higher quality Skinned Mesh Renders, with up to 4 bone weights per vertex, which reveals the weight painting problem. 

In the next build I've added mesh processing buttons which let you strip bone weights in the mesh to particular number of bones (I've confirmed that this fixes the stray vertex in Neos as well), but it's at the cost of lower quality mesh. I recommend fixing the weight paints in external program and re-importing instead.
